test/uk/gov/hmrc/economiccrimelevyregistration/controllers/actions/FakeDataRetrievalAction.scala

```diff
@@ -23,16 +23,25 @@ import uk.gov.hmrc.economiccrimelevyregistration.models.requests.{AuthorisedRequ
 
 import scala.concurrent.{ExecutionContext, Future}
 
-class FakeDataRetrievalAction(data: Registration, additionalInfo: Option[RegistrationAdditionalInfo] = None)
-    extends DataRetrievalAction {
+class FakeDataRetrievalAction(
+  registration: Registration,
+  registrationAdditionalInfo: Option[RegistrationAdditionalInfo] = None,
+  eclRegistrationReference: String
```

Comment:
Can't you default that to `testEclRegistrationReference`?

Reply:
Defaulting will prevent testing error cases when it is not present in the request

Reply:
This class doesn't mix in the EclTestData trait @paddynski-opencast, and i don't think it should.